### PR TITLE
Add subnet service

### PR DESCRIPTION
**Pull request by calvix**
towards https://github.com/giantswarm/roadmap/issues/491

### CODE REVIEW

**Comment by calvix** [comment]
its first draft, the idea is that operator is watching AWSMachinePool CR
 once it sees a CR it will check if the pool has an assigned a CIDR block if not it will get one that is not used and save it as an annotation on the CR, then it will add VPCCidrAssociation to the cluster VPC, and then it will adjust AWSCluster.Spec.Subnets to include the new subnets. cluster-api-provider-aws will create the subnets with proper TAGs 
 
 once the upstream PR is merged the ASG will be created with proper subnets

**Comment by calvix** [comment]
just looking for comments and opinions

File: main.go [hunk 1]
```diff
@@ -58,8 +57,8 @@ func main() {
 	var metricsAddr string
 	var enableLeaderElection bool
 	var probeAddr string
-	flag.StringVar(&defaultCidrRange, "default-cidr-range", "10.10.0.0/16","The default range which will be used to pick  free range for the subnets")
-	flag.StringVar(&defaultSubnetSize, "default-subnet-size","24", "Default size of subnet prefix for each AZ")
+	flag.StringVar(&defaultCidrRange, "default-cidr-range", "10.10.0.0/16", "The default range which will be used to pick  free range for the subnets")
```
Comment (AverageMarcus) [inline main.go]:
```suggestion
	flag.StringVar(&defaultCidrRange, "default-cidr-range", "10.10.0.0/16", "The default range which will be used to pick free range for the subnets")
```

File: pkg/key/key.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	return &awsClusterList.Items[0], nil
+}
+
+func HasCapiWatchLabel(labels map[string]string) bool {
+	value, ok := labels[ClusterWatchFilterLabel]
+	if ok {
+		if value == "capi" {
+			return true
+		}
+	}
```
Comment (AverageMarcus) [inline pkg/key/key.go]:
```suggestion
	if ok && value == "capi" {
		return true
	}
```

File: pkg/key/key.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+			Resource: "namespace",
+			Group:    v1.SchemeGroupVersion.Group,
+			Version:  v1.SchemeGroupVersion.Version,
+		},
+	}
+	kl, err := kubelock.New(c)
+	if err != nil {
+
+		return nil, err
+	}
```
Comment (AverageMarcus) [inline pkg/key/key.go]:
```suggestion
	if err != nil {
		return nil, err
	}
```

**Review by AverageMarcus — APPROVED** [review]
Looks good to me 👍  

Just some nitpick suggestions

**Pull request merged**